### DISCUSSION TITLE
Create image ubuntu-latest-minimal or similar

### DISCUSSION THREAD

**Discussion by cep21** [actions/virtual-environments#3242] [Ideas]
Hi,

Lots of people are running short on disk space running github actions.  This is creating hacks like this.   https://github.com/ThewBear/free-actions  You can see lots of them in github if you search for the rm string.

https://github.com/search?l=YAML&q=rm+%2Fusr%2Fshare%2Fdotnet&type=Code

They are often look like this: https://github.com/VirtusLab/intellij-pants-plugin-tests/blob/1b0d2d685e80dfa6ec59939e0d3c832a7f8e4f15/.github/workflows/prepare-image.yml#L15

It seems like a hack to `rm` the directories that happen to be taking a lot of space, and probably something that's not consistent from one build to the next in the future.  It may be worth a way to specify a minimal version to use, for people that need the extra disk space.

Some more follow up links here:

* https://github.com/ThewBear/free-actions
* https://github.com/actions/virtual-environments/issues/709
* https://github.com/NickleDave/vak/issues/341
* https://github.com/flashlight/wav2letter/actions/runs/74797824/workflow
* https://github.com/search?q=%22rm+-rf+%2Fusr%2Fshare%2Fdotnet%22&type=code

**Comment by AlenaSviridenko** [comment]
Hi @cep21,
you are just right in time with this idea 🙂 Glad to say, that we have already started preparing a plan to provide minimal Ubuntu images that will target a certain development stack. At this moment, I cannot provide any defined ETA because it is just a planning stage, but we see the issue for sure and looking for the solution.

**Reply by oittaa** [reply]
Any news on this one? Also, are there any plans for different distributions like Alpine Linux?

**Reply by miketimofeev** [reply]
The decision was made to focus on other stuff, so this is not something we could introduced soon

**Comment by crohr** [comment]
Hey @cep21, I know this issue is quite old, but if you're still looking for a solution, you could try an alternate runner such as https://github.com/runs-on/runs-on, which allows specifying disk size at runtime.

For instance for a 100GB disk:

```
runs-on: runs-on,runner=16cpu-linux,hdd=100
```
